File: biblio-boot/src/main/java/biblio_boot/service/AuteurService.java
package biblio_boot.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import biblio_boot.dao.AuteurDao;
import biblio_boot.model.Auteur;
import biblio_boot.model.Livre;

@Service
public class AuteurService {

	@Autowired
	AuteurDao auteurDao;

	public List<Auteur> getAllAuteurs() {
		return auteurDao.findAll();
	}

	public Auteur getById(Integer id)
	{
		return auteurDao.findById(id).orElse(null);
	}

	public Auteur insert(Auteur auteur)
	{
		return auteurDao.save(auteur);
	}
	
	public Auteur update(Auteur auteur)
	{
		return auteurDao.save(auteur);
	}

	public void delete(Integer id)
	{
		auteurDao.deleteById(id);
	}
}

File: biblio-angular/src/app/service/auteur-service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Service } from '@angular/core';
import { Observable } from "rxjs";
import { Auteur } from '../model/auteur';

@Service()
export class AuteurService {
    private http: HttpClient = inject(HttpClient);
    private apiUrl: string = '/auteur';

    public findAll(): Observable<Auteur[]> {
        return this.http.get<Auteur[]>(this.apiUrl);
    }

    public add(auteur: Auteur): Observable<Auteur> {
        return this.http.post<Auteur>(this.apiUrl, auteur);
    }

    public update(auteur: Auteur): Observable<Auteur> {
        return this.http.put<Auteur>(`${this.apiUrl}/${auteur.id}`, auteur);
    }

    public remove(auteur: Auteur): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${auteur.id}`);
    }

}
